src/android/io/sentry/SentryCordova.java

```diff
@@ -1,47 +1,370 @@
+        }
+        if (jsonOptions.has("dist") && jsonOptions.getString("dist") != null) {
+          options.setDist(jsonOptions.getString("dist"));
+        }
+        if (jsonOptions.has("enableAutoSessionTracking")) {
+          options.setEnableSessionTracking(jsonOptions.getBoolean("enableAutoSessionTracking"));
+        }
+        if (jsonOptions.has("sessionTrackingIntervalMillis")) {
+          options.setSessionTrackingIntervalMillis(jsonOptions.getInt("sessionTrackingIntervalMillis"));
+        }
+        if (jsonOptions.has("enableNdkScopeSync")) {
+          options.setEnableScopeSync(jsonOptions.getBoolean("enableNdkScopeSync"));
+        }
+        if (jsonOptions.has("attachStacktrace")) {
+          options.setAttachStacktrace(jsonOptions.getBoolean("attachStacktrace"));
+        }
+        if (jsonOptions.has("attachThreads")) {
+          // JS use top level stacktraces and android attaches Threads which
+          // hides them so by default we hide.
+          options.setAttachThreads(jsonOptions.getBoolean("attachThreads"));
+        }
 
-          SentryAndroid.init(this.cordova.getActivity().getApplicationContext(), options -> {
-            options.setDsn(dsn);
-          });
+        if (jsonOptions.has("enableNativeCrashHandling") && !jsonOptions.getBoolean("enableNativeCrashHandling")) {
+          final List<Integration> integrations = options.getIntegrations();
+          for (final Integration integration : integrations) {
+            if (integration instanceof UncaughtExceptionHandlerIntegration || integration instanceof AnrIntegration || integration instanceof NdkIntegration) {
+              integrations.remove(integration);
+            }
+          }
+        }
 
-          callbackContext.sendPluginResult(new PluginResult(Status.OK, true));
-          break;
-        default:
-          callbackContext.sendPluginResult(new PluginResult(Status.ERROR, "not implemented"));
-          break;
+        logger.info(String.format("Native Integrations '%s'", options.getIntegrations().toString()));
+        sentryOptions = options;
+      } catch (JSONException e) {
+        logger.info("Error parsing options JSON sent over native bridge.");
+        callbackContext.sendPluginResult(new PluginResult(Status.ERROR, false));
       }
+    });
 
-      return true;
+    callbackContext.sendPluginResult(new PluginResult(Status.OK, true));
+  }
+
+  private void captureEnvelope(String envelope, final CallbackContext callbackContext) {
+    try {
+      File installation = new File(sentryOptions.getOutboxPath(), UUID.randomUUID().toString());
+      try (FileOutputStream out = new FileOutputStream(installation)) {
+        out.write(envelope.getBytes(Charset.forName("UTF-8")));
+      }
+    } catch (Exception e) {
+      logger.info("Error reading envelope");
+      callbackContext.sendPluginResult(new PluginResult(Status.ERROR, false));
+    }
+
+    callbackContext.sendPluginResult(new PluginResult(Status.OK, true));
+  }
```

Comment:
on Flutter we do some validations to avoid broken envelopes:

https://github.com/getsentry/sentry-dart/blob/main/flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt#L121-L133